File: week8/javascript/Siby1lA/_4659.js
//00:16:00
function solution(case1) {
  let input = case1.split("\n").map((val) => val.trim());
  for (let i = 0; i < input.length - 1; i++) {
    const str = input[i];
    const vowels = ["a", "e", "i", "o", "u"];
    let bool = false;
    for (let j = 0; j < vowels.length; j++) {
      if (str.includes(vowels[j])) bool = true;
    }
    const arr = [];
    for (let j = 0; j < str.length; j++) {
      if (vowels.includes(str[j])) arr.push(0);
      else arr.push(1);
    }
    let continuos = 1;
    for (let j = 0; j < arr.length; j++) {
      if (arr[j] === arr[j + 1]) continuos += 1;
      else continuos = 1;

      if (continuos === 3) {
        bool = false;
        break;
      }
    }
    for (let j = 0; j < str.length; j++) {
      if (str[j] === str[j + 1]) {
        if (str[j] !== "e" && str[j] !== "o") bool = false;
      }
    }
    if (bool === true) console.log("<" + str + "> is acceptable.");
    else console.log("<" + str + "> is not acceptable.");
  }
}
const case1 = `a
tv
ptoui
bontres
zoggax
wiinq
eep
houctuh
end
`;

solution(case1);

File: week8/java/src/Gawq/_4659.java
package Gawq;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.StringReader;
// 37m41s
public class _4659 {
    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new StringReader(src));
        StringBuilder sb = new StringBuilder();
        char[] vowles = {'a', 'e', 'i', 'o', 'u'};

        while (true) {
            String s = input.readLine();
            if (s.equals("end")) break;
            else {
                boolean acceptable = true;
                boolean hasVowel = false;
                int cntVowel = 0;
                int cntConsonant = 0;
                for (int i = 0; i < s.length(); i++) {
                    char cur = s.charAt(i);
                    boolean isVowel = false;
                    for (int j = 0; j < vowles.length; j++) {
                        if (cur == vowles[j]) {
                            isVowel = true;
                            hasVowel = true;
                            cntVowel++;
                            cntConsonant = 0;
                            break;
                        }
                    }
                    if(!isVowel) {
                        cntConsonant++;
                        cntVowel = 0;
                    }
                    if (i == s.length() - 1) {
                        if (!hasVowel) {
                            sb.append("<" + s + "> is not acceptable.").append("\n");
                            acceptable = false;
                            break;
                        }
                    }
                    if (i >= 1) {
                        if (cur == s.charAt(i - 1) && cur != 'e' && cur != 'o') {
                            sb.append("<" + s + "> is not acceptable.").append("\n");
                            acceptable = false;
                            break;
                        } else if (cntVowel >= 3 || cntConsonant >= 3) {
                            sb.append("<" + s + "> is not acceptable.").append("\n");
                            acceptable = false;
                            break;
                        }
                    }
                }
                if (acceptable) sb.append("<" + s + "> is acceptable.").append("\n");
            }
        }
        System.out.println(sb);
    }

    private static String src = "a\n" +
            "tv\n" +
            "ptoui\n" +
            "bontres\n" +
            "zoggax\n" +
            "wiinq\n" +
            "eep\n" +
            "houctuh\n" +
            "end";
}
